monai/apps/deepedit/transforms.py

```diff
@@ -120,51 +531,401 @@ def find_guidance(self, discrepancy):
+        meta_dict_key = self.meta_keys or f"{self.ref_image}_{self.meta_key_postfix}"
+        if meta_dict_key not in d:
+            raise RuntimeError(f"Missing meta_dict {meta_dict_key} in data!")
+        if "spatial_shape" not in d[meta_dict_key]:
+            raise RuntimeError('Missing "spatial_shape" in meta_dict!')
+
+        # Assume channel is first and depth is last CHWD
+        original_shape = d[meta_dict_key]["spatial_shape"]
+        current_shape = list(d[self.ref_image].shape)[1:]
+
+        # in here we assume the depth dimension is in the last dimension of "original_shape" and "current_shape"
+        factor = np.array(current_shape) / original_shape
+
+        # Creating guidance for all clicks
+        all_guidances = dict()
+        for key_label in self.label_names.keys():
+            clicks = d.get(key_label, [])
+            clicks = list(np.array(clicks).astype(int))
+            all_guidances[key_label] = self._apply(clicks, factor)
+        d[self.guidance] = all_guidances
+        return d
+
+
+class ResizeGuidanceMultipleLabelCustomd(Transform):
+    """
+    Resize the guidance based on cropped vs resized image.
+    """
+
+    def __init__(self, guidance: str, ref_image: str) -> None:
+        self.guidance = guidance
+        self.ref_image = ref_image
+
+    def __call__(self, data):
+        d = dict(data)
+        # Assume channel is first and depth is last CHWD
+        current_shape = d[self.ref_image].shape[1:]
+        original_shape = d["image_meta_dict"]["spatial_shape"]
+
+        # original_shape = np.roll(original_shape, 1)
+
+        factor = np.divide(current_shape, original_shape)
+        all_guidances = dict()
+        for key_label in d[self.guidance].keys():
+            guidance = (
+                np.multiply(d[self.guidance][key_label], factor).astype(int).tolist()
+                if len(d[self.guidance][key_label])
+                else []
+            )
+            all_guidances[key_label] = guidance
+
+        d[self.guidance] = all_guidances
+        return d
+
+
+class SplitPredsLabeld(MapTransform):
+    """
+    Split preds and labels for individual evaluation
+
+    """
+
+    def __init__(self, keys: KeysCollection, allow_missing_keys: bool = False):
+        super().__init__(keys, allow_missing_keys)
+
+    def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndarray]:
+        d: Dict = dict(data)
+        for key in self.key_iterator(d):
+            if key == "pred":
+                for idx, (key_label, _) in enumerate(d["label_names"].items()):
+                    if key_label != "background":
+                        d[f"pred_{key_label}"] = d[key][idx + 1, ...][None]
+                        d[f"label_{key_label}"] = d["label"][idx + 1, ...][None]
+            elif key != "pred":
+                logger.info("This is only for pred key")
+        return d
+
+
+class ToCheckTransformd(MapTransform):
```

Comment:
please consider replacing this one with the existing DataStatsd transform https://docs.monai.io/en/stable/transforms.html#datastats

Reply:
Agree! Thanks for this